iroha_data_model/src/lib.rs

```diff
@@ -692,17 +713,17 @@ pub mod account {
         /// Asset's in this `Account`.
         pub assets: AssetsMap,
         /// `Account`'s signatories.
-        pub signatories: Signatories,
-        /// Permissions tokens of this account.
-        permission_tokens: Permissions,
+        pub signatories: RwLock<Signatories>,
+        /// Permissions tokens of this account
+        pub permission_tokens: RwLock<Permissions>,
         /// Condition which checks if the account has the right signatures.
         #[serde(default)]
-        pub signature_check_condition: SignatureCheckCondition,
+        pub signature_check_condition: RwLock<SignatureCheckCondition>,
         /// Metadata of this account as a key-value store.
-        pub metadata: Metadata,
+        pub metadata: RwLock<Metadata>,
```

Comment:
Metadata is a map, so same question.

Reply:
It needs to be Ord here, but it wouldn't be efficient in `HashMap`